### ISSUE TITLE
RESOLVER_RESOLVE_VARIABLE _ERROR

### ISSUE THREAD

**Issue by gavinsui1106** [serverless/serverless#12471]
### Issue description

when i used sls4 to deploy my project,the RESOLVER_RESOLVE_VARIABLE_ERROR is happend in codebuild's log

command 
serverless deploy --stage {mystage} --param="division=${envDivision}"

serverless.yml
${self:custom.${opt:stage,self:provider.stage}.XXX}
${file(xxx/serverless_xxx${param:division}.yml)}

what's wrong with my command?

### Context

when i used sls4 to deploy my project,the RESOLVER_RESOLVE_VARIABLE_ERROR is happend in codebuild's log

command 
serverless deploy --stage {mystage} --param="division=${envDivision}"

serverless.yml
${self:custom.${opt:stage,self:provider.stage}.XXX}
${file(xxx/serverless_xxx${param:division}.yml)}

what's wrong with my command?

**Comment by gavinsui1106** [comment]
when i deleted the code "--param="division=${envDivision}"" , so that the deploy command is [serverless deploy --stage ${env}] then other error is happend. but when i use the same code in sls3, its successfully builded!!!

<error message >
   Failed to resolve variable 'custom.${opt:stage' with resolver 'config' :ERROR :Key "custom.${opt:stage" is not defined in config file.

<here is my whole code>
**buildspec.yml**
`serverless deploy --stage ${env}`
#${env} is set with environment during AWS's Coodebuild . for example ,Dev is the value of env.

**serverless.yml**
```
provider:
  name: test
  runtime: dotnet6
  region: XXX
  environment: ${file(slsconf/serverless_environment.yml)}

custom:
   ${file(slsconf/serverless_custom${opt: stage, self: provider.stage}.yml)}
```
**slsconf/serverless_environment.yml**
```
TZ: XXX
EnvironmentVariable1: ${self: custom.${opt: stage, self: provider.stage}.EnvironmentVariable1Val}
EnvironmentVariable2: ${self: custom.${opt: stage, self: provider.stage}.EnvironmentVariable2Val}
```
**slsconf/serverless_customDev.yml**
```
EnvironmentVariable1Val: XXX
EnvironmentVariable2Val: XXX
```

**Comment by gavinsui1106** [comment]
@yeyeric 
maybe this error is fixed in 4.0.30,but another error is happened.
